views-fieldset/src/main/java/io/micronaut/views/fields/elements/InputNumberFormElement.java

```diff
@@ -0,0 +1,229 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.views.fields.elements;
+
+import io.micronaut.core.annotation.Experimental;
+import io.micronaut.core.annotation.Introspected;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.views.fields.InputType;
+import io.micronaut.views.fields.messages.Message;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Input type Number.
+ * @see <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number">input number</a>
+ * @param name Name of the form control. Submitted with the form as part of a name/value pair
+ * @param id It defines an identifier (ID) which must be unique in the whole document
+ * @param value A number representing the value of the input number.
+ * @param max The maximum value to accept for this input.
+ * @param min The minimum value to accept for this input.
+ * @param placeholder The placeholder attribute is a string that provides a brief hint to the user as to what kind of information is expected in the field.
+ * @param required If true indicates that the user must specify a value for the input before the owning form can be submitted.
+ * @param readOnly A Boolean attribute which, if present, means this field cannot be edited by the user.
+ * @param step The step attribute is a number that specifies the granularity that the value must adhere to, or the special value any.
+ * @param label represents a caption for an item in a user interface
+ * @param errors errors associated with this input
+ *
+ * @author Sergio del Amo
+ * @since 4.1.0
+ */
+@Experimental
+@Introspected(builder = @Introspected.IntrospectionBuilder(builderClass = InputNumberFormElement.Builder.class))
+public record InputNumberFormElement(@NonNull String name,
+                                     @Nullable String id,
+                                     @Nullable Number value,
+                                     @Nullable Number max,
+                                     @Nullable Number min,
+                                     @Nullable String placeholder,
+                                     boolean required,
+                                     boolean readOnly,
+                                     @Nullable String step,
+                                     @NonNull Message label,
+                                     @NonNull List<Message> errors) implements InputFormElement, GlobalAttributes, FormElementAttributes {
+
+    @Override
+    @NonNull
+    public String getType() {
+        return InputType.ATTR_TYPE_NUMBER;
+    }
+
+    /**
+     *
+     * @return the Input Number FormElement Builder
+     */
+    @NonNull
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Input Number Builder.
+     */
+    public static class Builder {
```

Comment:
can be `final`

Reply:
resolved via resolved via https://github.com/micronaut-projects/micronaut-views/pull/613/commits/56fc641315f8db4d2471cac1d3dbfebbbe374488